### PR TITLE
unpin polars

### PR DESCRIPTION
**Pull request by alangenfeld**
use `"delta_write_options": {"engine": "pyarrow"},` in partition tests 

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
* **#23708** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23708?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alangenfeld and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-polars/setup.py [hunk 1]
```diff
@@ -38,13 +38,13 @@ def get_version() -> str:
...
         f"dagster{pin}",
-        "polars>=0.20.0,<1.3.0",
+        "polars>=0.20.0",
         "pyarrow>=8.0.0",
         "typing-extensions>=4.7.0",
         "universal_pathlib>=0.1.4",
     ],
     extras_require={
-        "deltalake": ["deltalake>=0.15.0"],
+        "deltalake": ["deltalake>=0.15.0,<0.19.0"],
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-polars/setup.py:47]: test failures that motivated pin:

```

[2024-08-16T16:18:46Z] FAILED dagster_polars_tests/test_polars_delta.py::test_polars_delta_native_partitioning - ValueError: Partition filters can only be used with PyArrow engine, use predicate instead. PyArrow engine will be deprecated in 1.0
--
  | [2024-08-16T16:18:46Z] FAILED dagster_polars_tests/test_polars_delta.py::test_polars_delta_native_multi_partitions - ValueError: Partition filters can only be used with PyArrow engine, use predicate instead. PyArrow engine will be deprecated in 1.0
  | [2024-08-16T16:18:46Z] FAILED dagster_polars_tests/test_polars_delta.py::test_polars_delta_native_partitioning_loading_single_partition - ValueError: Partition filters can only be used with PyArrow engine, use predicate instead. PyArrow engine will be deprecated in 1.0


```

Comment (alangenfeld) [reply python_modules/libraries/dagster-polars/setup.py:47]: i believe this is what is listed in the breaking changes section here:
https://github.com/delta-io/delta-rs/releases/tag/python-v0.19.0

**Comment by danielgafni** [comment]
I think we can do this instead: 

delta_write_options={"writer_engine": "pyarrow"}

Pyarrow was the default engine previously. 

I can look into changing this setting on our side later.

**Comment by alangenfeld** [comment]
ok thanks that was the pointer i needed to fix the tests and get the build green

**Review by rexledesma — APPROVED** [review]

**Comment by ion-elgreco** [comment]
> I think we can do this instead: 
> 
> delta_write_options={"writer_engine": "pyarrow"}
> 
> Pyarrow was the default engine previously. 
> 
> I can look into changing this setting on our side later. 

I would honestly just drop the delta support in dagster-polars to not maintain extra stuff.

I releases some improved forks of dagster-deltalake implementation https://github.com/ASML-Labs/dagster-delta

**Pull request merged**